.github/workflows/build-vcpkg.yml

```diff
@@ -21,348 +51,195 @@ on:
-
-      - name: OSX Dependencies
-        if: ${{ contains(matrix.os, 'macos') && contains(matrix.event_name, github.event_name) && needs.preamble.outputs.platform }}
-        shell: "bash"
-        run: |
-          brew install bison flex pkg-config autoconf autoconf-archive automake libtool cmake openjdk@11
-
-      - name: "Remove builtin vcpkg (old)"
-        if: ${{ contains(matrix.event_name, github.event_name) && needs.preamble.outputs.platform }}
-        working-directory: .
-        shell: "bash"
-        run: |
-          ${{ matrix.sudo }} rm -rf "$VCPKG_INSTALLATION_ROOT"
-
-      - name: Checkout HPCC-Platform
-        if: ${{ contains(matrix.event_name, github.event_name) && needs.preamble.outputs.platform }}
-        uses: actions/checkout@v3
-        with:
-          ref: ${{ needs.preamble.outputs.community_ref }}
-          submodules: recursive
-          path: ./HPCC-Platform
-
-      - name: "vcpkg Bootstrap"
-        if: ${{ contains(matrix.event_name, github.event_name) && needs.preamble.outputs.platform }}
-        shell: "bash"
-        run: |
-          ./HPCC-Platform/vcpkg/bootstrap-vcpkg.sh
-
-      - name: "Setup NuGet credentials"
-        if: ${{ contains(matrix.event_name, github.event_name) && needs.preamble.outputs.platform }}
-        shell: "bash"
-        run: |
-          ${{ matrix.mono }} `./HPCC-Platform/vcpkg/vcpkg fetch nuget | tail -n 1` \
-            sources add \
-            -name "GitHub" \
-            -source "https://nuget.pkg.github.com/hpcc-systems/index.json" \
-            -storepasswordincleartext \
-            -username "${{ github.actor }}" \
-            -password "${{ secrets.GITHUB_TOKEN }}"
-
-      - name: CMake Packages
-        if: ${{ contains(matrix.event_name, github.event_name) && needs.preamble.outputs.platform }}
-        shell: "bash"
-        run: |
-          mkdir -p ./build
-          cmake -S ./HPCC-Platform -B ./build ${{ matrix.cmake_config_options }}
-          cmake --build ./build ${{ matrix.cmake_build_options }}
-
-      - name: Upload error logs
-        if: ${{ failure() || cancelled() }}
-        uses: actions/upload-artifact@v3
-        with:
-          name: ${{ matrix.os }}--${{ matrix.ln }}-${{ matrix.container }}-logs
-          path: ./build/**/*.log
+  build-workflow-dispatch:
+    if: ${{ contains('workflow_dispatch', github.event_name) }}
+    uses: ./.github/workflows/build-docker.yml
+    with:
+      os: ${{ inputs.os }}
+      ln: ${{ inputs.ln }}
+      upload-package: true
+      asset-name: 'docker-package'
+    secrets: inherit
+
+  test-workflow-dispatch:
+    if: ${{ contains('workflow_dispatch', github.event_name) && inputs.smoketest == true }}
+    needs: build-workflow-dispatch
+    uses: ./.github/workflows/test-smoke-gh_runner.yml
+    with:
+      os: ${{ inputs.os }}
+      asset-name: 'docker-package'
+    secrets: inherit
+
+  build-docker-ubuntu-23_10:
+    if: ${{ contains('pull_request,push', github.event_name) }}
```

Comment:
Remove ,push from this entire file (was only used for testing on my forked repo)

Reply:
Can leave the ",push" and remove the global push trigger (as its handy for local testing)

Reply:
TODO